File: data-structures-and-algorithms/sliding-window/easy/maximum-average-subarray-i/maximum_average_subarray1.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        return solution(nums, k)

def solution(nums, k):
    windowSum = sum(nums[i] for i in range(k))
    maxSum = windowSum
    for r in range(k, len(nums)):
        windowSum += nums[r]
        windowSum -= nums[r-k]
        maxSum = max(maxSum, windowSum)

    return maxSum/k

File: data-structures-and-algorithms/sliding-window/easy/maximum-average-subarray-i/maximumAverageSubarray1.ts
function findMaxAverage(nums: number[], k: number): number {
  let windowSum = 0

  // Calculate the sum of the first window
  for (let i = 0; i < k; i++) {
    windowSum += nums[i]
  }

  let maxSum = windowSum

  // Slide the window across the array
  for (let i = k; i < nums.length; i++) {
    windowSum += nums[i] - nums[i - k]
    maxSum = Math.max(maxSum, windowSum)
  }

  return maxSum / k
};